src/plugins/layout.ts

```diff
@@ -162,14 +159,13 @@ export default class Layout implements IPlugin {
         type: this.eventName,
         state: eventState
       });
-      this.layoutStates[eventState.index] = eventState;
```

Comment:
Why remove this ?

Reply:
Because we are caching states in state provider, and not layout. The benefit if we can walk the relationship between states (parent, etc.) without involving an external module like Layout. We can discuss this more offline if you would like.